testsuite/test_ta_workflow.py

```diff
@@ -275,6 +275,44 @@ def test_ta_edit_assignment(STC, client, base_db_setup):
     assert new_assignment == assignment
 
 
+@pytest.mark.django_db
+def test_soft_deadline_change_trigger_recompute(client, base_db_setup, mock_config):
+    (_, ta, user, course, assignment) = base_db_setup
+
+    mock_config.start_server()
+
+    submission = assignment.submission_set.create(
+        score=100.00, state=Submission.STATE_DONE, user=user,
+    )
+
+    assert submission.penalty == 0
+
+    client.login(username=ta.username, password="pw")
+
+    deadline_soft = datetime.now() - timedelta(days=365)
```

Comment:
To avoid dealing with holidays, I just used a full year and applied the maximum penalty.